gnssanalysis/filenames.py

```diff
@@ -722,6 +753,35 @@ def determine_name_props_from_filename(filename: str) -> Dict[str, Any]:
     }
 
 
+def warn_on_unexpected_filename(input_file: pathlib.Path) -> bool:
+    expected_file_name = determine_file_name(input_file, defaults={}, overrides={})
+    if input_file.name != expected_file_name:
+        logging.warning(f"File name: '{input_file.name}' "
+                       f"didn't match expected: '{expected_file_name}'. "
+                       "Contents may be incorrect and lead to failures."
+                       )
+        return False
+    return True
+
+
+def check_file_timespan_as_claimed(input_file: pathlib.Path) -> bool:
+    try:
+        actual_timespan:datetime.timedelta = determine_file_props(
+                                                                 file_path=input_file, defaults={}, overrides={}
+                                                                 )["timespan"]
+    except NotImplementedError as e:
+        logging.warning("Couldn't check file timespan. Format not yet supported by determine_file_props(): " + str(e))
+        return False # Assume it's bad, as we can't verify it's good.
+    claimed_timespan:datetime.timedelta = determine_name_props_from_filename(input_file.name)["timespan"]
+    
+    if claimed_timespan != actual_timespan:
```

Comment:
Similar to the above, I feel slightly uncomfortable around the logging messages in library code and the assumptions made for unsupported files. These have the feeling of being domain/application specific. If you then remove that logging and catching, this function boils down to a slightly nicer `return determine_file_props(input_file)["timespan"] == determine_name_props_from_filename(input_file.name)["timespan"]` and I'm not sure whether that's justifying its existence?

Reply:
Updated, see above comment.